### DISCUSSION TITLE
Dyanmic 3DGS: How to add custom buttons to the NeRF studio viewer for my own method?

### DISCUSSION THREAD

**Discussion by josephldobson** [nerfstudio-project/nerfstudio#3458] [Q&A] [answered]
I've been writing a dynamic gaussian splatting method as a NeRF Studio method and I would like to add some new buttons to the viewer for this method only. For example:

- Pause dynamic playback
- Scrub through time
- ...

Whats the best/easiest way to do this without breaking anything?
Thanks!

**Comment by maturk** [comment, marked as answer]
Hi @josephldobson , there are documentation [here](https://docs.nerf.studio/developer_guides/viewer/custom_gui.html). In short, you can add buttons in the `populate_modules(self)` (or `__init__(self)`) function by importing viewer_elements and adding ur needed buttons.

**Reply by josephldobson** [reply]
Thanks for quick reply, I couldn't find this before
